auto_nag/scripts/reminder.py

```diff
@@ -0,0 +1,79 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import re
+
+import dateutil.parser
+
+from auto_nag.bzcleaner import BzCleaner
+
+DATE_MATCH = re.compile(r"\[reminder-?[a-zA-Z0-9]* ([0-9]{4}-[0-9]{1,2}-[0-9]{1,2})\]")
+
+
+class Reminder(BzCleaner):
+    def __init__(self):
+        super(Reminder, self).__init__()
+        self.changes_per_bug = {}
+
+    def description(self):
+        return "Allow users to set reminders on bugs"
+
+    def get_bz_params(self, date):
+        params = {
+            "include_fields": ["assigned_to", "whiteboard", "triage_owner"],
+            "f1": "whiteboard",
+            "o1": "substring",
+            "v1": "[reminder",
+        }
+
+        return params
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        new_whiteboard = whiteboard = bug["whiteboard"]
+        matches = DATE_MATCH.findall(whiteboard)
+        for m in matches:
+            target = dateutil.parser.parse(m)
+            today = datetime.datetime.today()
+
+            if target.date() < today.date():
+                replace_string = f"[reminder {m}]"
+                new_whiteboard = new_whiteboard.replace(replace_string, "")
+
+        if new_whiteboard != whiteboard:
+            assignee = bug["assigned_to"]
+            triage_owner = bug["triage_owner"]
+
+            self.changes_per_bug[bugid] = {
+                "comment": {"body": self.comment_tpl_for_bugid(bugid)},
+                "whiteboard": new_whiteboard,
+                "flags": [
+                    {
+                        "name": "needinfo",
+                        "requestee": assignee,
+                        "status": "?",
+                        "new": "true",
+                    },
+                    {
+                        "name": "needinfo",
+                        "requestee": triage_owner,
+                        "status": "?",
+                        "new": "true",
+                    },
+                ],
+            }
+
+        return bug
+
+    def get_autofix_change(self):
+        return self.changes_per_bug
+
+    def comment_tpl_for_bugid(self):
+        return "I am placing a needinfo on the bug per the reminder whiteboard tag; please refer to the comment where the tag was placed to understand the reason."
```

Comment:
We could add `history` to `include_fields`; this will fetch the history in the same round.

https://github.com/mozilla/relman-auto-nag/blob/032d00b21faebc2cc71ab8d57ebc6194b5185b24/auto_nag/scripts/reminder.py#L26